File: offer-search-service/src/main/java/pl/allezon/offersearch/repository/OfferRepository.java
package pl.allezon.offersearch.repository;

import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;
import org.springframework.stereotype.Repository;
import pl.allezon.offersearch.domain.Offer;

import java.util.List;

@Repository
public interface OfferRepository extends ElasticsearchRepository<Offer, Long> {

    List<Offer> findByNameOrDescription(String name, String description);
}

File: offer-search-service/src/main/java/pl/allezon/offersearch/repository/OfferRepository.kt
package pl.allezon.offersearch.repository

import org.springframework.data.elasticsearch.repository.ElasticsearchRepository
import org.springframework.stereotype.Repository
import pl.allezon.offersearch.domain.Offer

@Repository
interface OfferRepository : ElasticsearchRepository<Offer, Long> {

    fun findByNameOrDescription(name: String, description: String): List<Offer>

}
